### DISCUSSION TITLE
How to clear task execution history?

### DISCUSSION THREAD

**Discussion by lilws** [Flexget/Flexget#3101] [Q&A]
I let flexget run for couple months and lately when I tried to access flexget it loading a huge entry of Task - Latest execution. Today I just delete all database manually.

I wonder is there anyway to clear these task execution in flexget? I tried to Cleanup and vacuum but these entries still there. Last time it is more than 50k entries and loading very slow.

![image](https://user-images.githubusercontent.com/12478478/127502466-2ef66371-9e71-40fa-988e-7b07b2041e57.png)

**Comment by paranoidi** [comment]
Unfortunately I am painfully aware that there is no cleanup for task execution history as I left that implementation to be done later. Years ago. This makes even executing `flexget status` from cli slow ...

**Reply by lilws** [reply]
It's nice that you are aware of this. It should implements days, weeks or months to keep these entries or clean up this.

For now I use SQLite browser to access DB when flexget is slow to clear old entries.

**Comment by lilws** [comment]
Ok, after a while, I just figured that I can reset DB plugins status to remove all the execution tasks.
But it would be nice if this can run automatically every weeks or so.

**Reply by 4oo4** [reply]
@lilws I'm facing a similar problem. When you did that were you using a sqlite browser, or did you do that with `flexget reset-plugin $plugin` ? And what plugins did you specifically clear?

cheers

**Reply by lilws** [reply]
@4oo4 I use the flexget reset-plugins. It's "status".

**Comment by gazpachoking** [comment]
We have a database cleanup event, if somebody wants to write a PR which cleans up entries after some period of time it would be appreciated.

**Reply by 4oo4** [reply]
So essentially just calling this on an interval, inside different plugins?

```python
@event('manager.db_cleanup')
```

**Reply by gazpachoking** [reply]
A function with that decorator will automatically be called periodically. It then just has to implement the cleanup within there, say, deleting all the database entries older than a year, or some interval that seems good. If you search for that line in the source you can see how other plugins are handling it.

**Reply by gazpachoking** [reply]
Oh, it looks like there is some cleanup already. Anything over a year should be automatically cleaned up. https://github.com/Flexget/Flexget/blob/1a394627ce54d2b6cf771dc13e9240db70ead5e8/flexget/components/status/status.py#L59-L74

**Reply by gazpachoking** [reply]
I'm guessing we could add some to that cleanup, maybe a max number of runs of history per task?
